lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -792,6 +803,22 @@ class InferenceMapping {
 
 } // namespace
 
+/// Check if a FIRRTL type contains any uninferred widths.
+static bool hasUninferredWidth(Type type) {
+  return TypeSwitch<Type, bool>(type)
+      .Case<FlipType, FVectorType>(
+          [](auto type) { return hasUninferredWidth(type.getElementType()); })
+      .Case<BundleType>([](auto type) {
+        for (auto fieldType : type.getElements())
+          if (hasUninferredWidth(fieldType.type))
+            return true;
+        return false;
+      })
+      .Case<SIntType, UIntType, AnalogType>(
+          [](auto type) { return !type.hasWidth(); })
+      .Default([](auto) { return false; });
+}
```

Comment:
This needs to be cached, otherwise it will be O(total-size-of-type) which will explode compile time.  This may make sense to be a bit cached on FIRRTL type itself (getRecursiveTypeProperties), just like `isPassive` is.

Reply:
Good point! I'll move this over to `FIRRTLType` itself.